### DISCUSSION TITLE
Can not import anything to commands.ts file

### DISCUSSION THREAD

**Discussion by Illyuha1705** [cypress-io/cypress#26281] [Questions and Help]
In my angular project I'm using Cypress. There is a need to import usual function into commands.ts file. Of course, commands.ts has custom commands. The problem is with eslint. After importing something, all custom commands has an issue: Argument of type '"customCommandName"' is not assignable to parameter of type 'keyof Chainable '. Also I have Cypress namespace with Chainable interface, where I'm declaring custom commands. So without imports all works fine

![image](https://user-images.githubusercontent.com/75776668/228499335-2fc7c460-6748-4789-83a5-ca96f38e9bcf.png)

Step to reproduce: 
- have installed cypress,
- have al least 1 custom command
- have eslint recommended, cypress-eslint, typescript-eslint recommended plugins
- import something to commands.ts file

**Comment by portyu9** [comment]
Hi @Illyuha1705 

The import changes `commands.ts` from an ambient TypeScript script into a **module**. Once that happens, this no longer augments Cypress's global namespace:

```ts
declare namespace Cypress {
  interface Chainable {
    login(email: string, password: string): void
  }
}
```

For a file containing imports, wrap the declaration in `declare global`:

```ts
declare global {
  namespace Cypress {
    interface Chainable {
      login(email: string, password: string): Chainable<void>
    }
  }
}
```

Then your implementation can remain:

```ts
Cypress.Commands.add('login', (email, password) => {
  // ...
})
```

If you move the typings to a standalone `.d.ts` file containing **no imports or exports**, then the original `declare namespace Cypress` form is valid there.

Also, in the screenshot you are importing:

```ts
import a from 'cypress/e2e/approve-vacation.cy'
```

I would avoid importing a spec file from `commands.ts`. Spec files can have test-registration side effects. Extract whatever you need into a normal helper/support module and import that instead.

This behavior is still the same in current Cypress: files with imports need `declare global { namespace Cypress { ... } }` to augment `Chainable`.

Please mark this answer as accepted if it helped, thank you.
